### PR TITLE
feat(linter): eslint-plugin-jest/prefer-lowercase-title

### PR DESCRIPTION
**Pull request by eryue0220**
part of https://github.com/oxc-project/oxc/issues/492

Rule Detail: [link](https://github.com/jest-community/eslint-plugin-jest/blob/main/src/rules/prefer-lowercase-title.ts)

### CODE REVIEW

**Comment by eryue0220** [comment]
I'll back to check the performance problem.

**Comment by mysteryven** [comment]
I think the performance problem is not related to this PR, I'll take a close look at this PR several hours later!

**Review by mysteryven — COMMENTED** [review]
LGTM, only two nitpick.

File: crates/oxc_linter/src/rules/jest/prefer_lowercase_title.rs [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Some(ParsedJestFnCallNew::GeneralJestFnCall(jest_fn_call)) =
+            parse_jest_fn_call(call_expr, possible_jest_node, ctx)
+        else {
+            return;
+        };
+
+        if ignores.contains(&jest_fn_call.name.to_string()) {
```
[the remark this answers is not in the document]
Comment (eryue0220) [reply crates/oxc_linter/src/rules/jest/prefer_lowercase_title.rs]: Thanks, updated~

File: crates/oxc_linter/src/rules/jest/prefer_lowercase_title.rs [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+        }
+
+        if ignore.contains(&it_name.to_string()) {
+            ignores.extend(
+                test_case_name
+                    .iter()
+                    .filter(|alias| alias.ends_with(it_name))
+                    .map(|alias| (*alias).to_string()),
+            );
+        }
```
[the remark this answers is not in the document]
Comment (eryue0220) [reply crates/oxc_linter/src/rules/jest/prefer_lowercase_title.rs]: updated.

**Comment by mysteryven** [comment]
Some test cases are broken.

**Review by mysteryven — APPROVED** [review]
Nice work, thank you!

**Pull request merged**